actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java

```diff
@@ -110,4 +119,122 @@ public void execute(JobExecutionContext ctx) throws JobExecutionException {
+    if (isCountIncrementStatus) {
+      courseBatchMap.put(JsonKey.COUNTER_INCREMENT_STATUS, true);
+      courseBatchMap.put(JsonKey.COUNT_INCREMENT_DATE, todayDate);
+    }
+    if (isCountDecrementStatus) {
+      courseBatchMap.put(JsonKey.COUNTER_DECREMENT_STATUS, true);
+      courseBatchMap.put(JsonKey.COUNT_DECREMENT_DATE, todayDate);
+      courseBatchMap.put(JsonKey.STATUS, ProjectUtil.ProgressStatus.COMPLETED.getValue());
+    }
+    courseBatchMap.put(JsonKey.UPDATED_DATE, ProjectUtil.getFormattedDate());
+  }
+
+  private void updateBatchMap(Map<String, Object> map) {
+    String courseId = (String) map.get(JsonKey.COURSE_ID);
+    String enrollmentType = (String) map.get(JsonKey.ENROLLMENT_TYPE);
+    if (JsonKey.OPEN.equals(enrollmentType)) {
+      if (openBatchMap.containsKey(courseId)) {
+        openBatchMap.get(courseId).add(map);
+      } else {
+        List<Map<String, Object>> batchList = new ArrayList<>();
+        batchList.add(map);
+        openBatchMap.put(courseId, batchList);
+      }
+    } else {
+      if (privateBatchMap.containsKey(courseId)) {
+        privateBatchMap.get(courseId).add(map);
+      } else {
+        List<Map<String, Object>> batchList = new ArrayList<>();
+        batchList.add(map);
+        privateBatchMap.put(courseId, batchList);
+      }
+    }
+  }
+
+  private void contentMapUpdate(boolean increment) {
+    for (Map.Entry<String, List<Map<String, Object>>> openBatchList : openBatchMap.entrySet()) {
+      String contentName = CourseBatchSchedulerUtil.getContentName(JsonKey.OPEN);
+      updateContentMapCount(
+          increment, contentName, openBatchList.getKey(), openBatchList.getValue().size());
+    }
+    for (Map.Entry<String, List<Map<String, Object>>> privateBatchList :
+        privateBatchMap.entrySet()) {
+      String contentName = CourseBatchSchedulerUtil.getContentName(JsonKey.INVITE_ONLY);
+      updateContentMapCount(
+          increment, contentName, privateBatchList.getKey(), privateBatchList.getValue().size());
+    }
+    updateEkstepAndDb();
+  }
+
+  private void updateContentMapCount(
+      boolean increment, String contentName, String courseId, int size) {
+    int val = (int) contentMap.get(courseId).getOrDefault(contentName, 0);
+    if (increment) {
+      val += size;
+    } else {
+      val -= size;
+      if (val < 0) {
+        val = 0;
+      }
+    }
+    contentMap.get(courseId).put(contentName, val);
+  }
+
+  private void getContentAndStore(Map<String, List<Map<String, Object>>> batchMap) {
+    Map<String, String> ekstepHeader = CourseBatchSchedulerUtil.headerMap;
+    for (Map.Entry<String, List<Map<String, Object>>> batchList : batchMap.entrySet()) {
+      String courseId = batchList.getKey();
+      if (!contentMap.containsKey(courseId)) {
+        Map<String, Object> ekStepContent =
+            CourseEnrollmentActor.getCourseObjectFromEkStep(courseId, ekstepHeader);
```

Comment:
Can be more efficient if - multiple courseIds can be fetched together...
can you confirm if that is possible.